tap_appstore/client.py

```diff
@@ -18,13 +25,128 @@
+        """Retrieve the configured start date, formatted as specified."""
+        start_date_str = self.config.get('start_date', default_date)
+        try:
+            return datetime.strptime(start_date_str, date_format)
+        except ValueError as e:
+            logger.error(f"Invalid start date format: {start_date_str}. Error: {e}")
+            return None
+
+    def setup_endpoint(self, start_date):
+        """Set up the endpoint for the API call. Override in subclass as needed."""
+        # Default to a generic endpoint configuration; specific streams will override this
+        return None
+
+    def parse_report_line(self, start_date):
+        """Parse a line returned by get_records"""
+        # Default to a generic configuration; specific streams will override this
+        return None
+
+    def increment_date(self, date):
+        """Increment date by one day. Override in subclass if different increment is needed."""
+        return date + timedelta(days=1)
+
+    def update_stream_state(self, date):
+        """Update the stream state with the new date."""
+        self.stream_state['start_date'] = date.strftime('%Y-%m-%d')
+        self.logger.info(f"Updating state, new start date is {self.stream_state['start_date']}")
+
+    def get_records(self, context: dict = None):
+        start_date = self.get_start_date(default_date='2024-01-01')
+        if not start_date:
+            logger.error("Start date could not be determined.")
+            return
+
+        start_date = start_date.replace(tzinfo=None)
+        date_limit = datetime.now().replace(tzinfo=None) - timedelta(days=2)
+
+        all_data = []
+        is_first_chunk = True
+
+        while start_date <= date_limit:
+            endpoint = self.setup_endpoint(start_date)
+            if not endpoint:
+                logger.error("Endpoint configuration is missing.")
+                break
+
+            try:
+                response = endpoint.get()
+
+                # Stream full TSV file as Gzip stream and load into memory
+                for chunk in response.iter_decompress():
+                    chunk_data = chunk.decode('utf-8')
+                    if not is_first_chunk:
+                        chunk_data = "\n".join(chunk_data.splitlines()[1:])
+                    all_data.append(chunk_data)
+
+                start_date = self.increment_date(start_date)
+                self.update_stream_state(start_date)
+            except Exception as e:
+                logger.error(f"Failed to process data: {str(e)}")
+                break
+
+        data_io = StringIO("\n".join(all_data))
+
+        # Rename column names from 'App Name' -> 'app_name'
+        first_line = data_io.readline().strip()
+        fieldnames = [col.strip().replace(' ', '_').lower() for col in first_line.split('\t')]
+
+        # Load TSV from memory
+        reader = csv.DictReader(data_io, delimiter='\t', fieldnames=fieldnames)
```

Comment:
Earlier we were just `yield`ing a list of fields for each record, but since we now load the TSV file, we can easily `yield` a dictionary of TSV headers as keys and row contents as values, which is what `get_records()` requires us to do - https://sdk.meltano.com/en/latest/classes/singer_sdk.RESTStream.html#singer_sdk.RESTStream.get_records

>     One item per (possibly processed) record in the API.

Reply:
This current `get_records()` implementation works with `meltano el tap-appstore target-jsonl`.

However, it fails with `meltano el tap-appstore target-parquet` due to some typing issues caused by data ending up in the wrong columns etc. Looking into that as I'm not sure if it's a problem with applaud or something else.

Reply:
With [d01d2bd](https://github.com/Automattic/tap-appstore-singer-sdk/pull/1/commits/d01d2bdeb59f8f98d0eaf485abd7a9b1339cf9c9) we are now testing [a forked / PyJWT dependency-fixed appstoreconnectapi Python module](https://github.com/Ponytech/appstoreconnectapi/compare/master...mahangu:appstoreconnectapi:fix-dependencies) for the API connection, as I was running into some issues with applaud and the data it was returning and wanted to see if `appstoreconnectapi` would be better.

We can revert this commit to test with applaud though.

Reply:
We are now using `appstoreconnectapi` instead of `applaud`.